### PR TITLE
Document automatic visibilty and metadata handling

### LINKED ISSUE #1047: [Metadata] Documentation
Document how visibility and metadata applicability are applied automatically once #1034 is in.

### PR DESCRIPTION
**Pull request by nguerrera**
Fix #1047

### CODE REVIEW

File: packages/website/src/docs/standard-library/http.md [hunk 1]
```diff
@@ -294,3 +298,134 @@ namespace Pets {
...
+```cadl
+@route("/users")
+interface Users {
+  @post create(...User): User;
+}
+```
+
+Abstractly, this expresses that a create operation that takes and returns a user. But concretely, at the HTTP protocol level, a create request and response look like this:
+
+```
```
Comment (timotheeguerin) [inline packages/website/src/docs/standard-library/http.md:375]: Think I need to fix something with the css of code blocks without a language.
image

**Review by timotheeguerin — APPROVED** [review]

File: packages/website/src/docs/standard-library/http.md [hunk 2]
```diff
@@ -142,19 +142,23 @@ namespace Pets {
...
-2. The set of properties of the return model that are not marked @header, @query, or @path form the response body.
+2. The set of properties of the return model that are not marked @header or @statusCode form the response body.
 3. If the return type is not a model, then it defines the response body.
 
 This is how we were able to return Pet and Pet[] bodies without using @body for list and read. We can actually write
 create in the same terse style by spreading the Pet object into the parameter list like this:
 
+See also [Advanced Metadata](#advanced-metadata) for more details.
+
+````cadl
```
[the remark this answers is not in the document]
Comment (nguerrera) [reply packages/website/src/docs/standard-library/http.md:153]: Oops, thought I fixed that, maybe I didn't push.

[the remark this answers is not in the document]
Comment (nguerrera) [reply packages/website/src/docs/standard-library/http.md:153]: copilot is going nuts trying to put ```cadl everywhere. 🤣

File: packages/website/src/docs/standard-library/http.md (hunk 1 above)
Comment (timotheeguerin) [reply packages/website/src/docs/standard-library/http.md:373]: Double box and it looks bad if overlflows, fixed here https://github.com/microsoft/cadl/pull/1136

Comment (nguerrera) [reply packages/website/src/docs/standard-library/http.md:373]: Got it and thanks!

**Pull request merged**